crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -245,18 +246,22 @@ impl Database for MongoDb {
         Ok(vec_jobs)
     }
 
-    async fn get_last_successful_job_by_type(&self, job_type: JobType) -> Result<Option<JobItem>> {
+    async fn get_latest_job_by_type_and_status(
+        &self,
+        job_type: JobType,
+        job_status: JobStatus,
+    ) -> Result<Option<JobItem>> {
         let filter = doc! {
             "job_type": bson::to_bson(&job_type)?,
-            "job_status": bson::to_bson(&JobStatus::Completed)?
+            "job_status": bson::to_bson(&job_status)?
         };
         let find_options = FindOneOptions::builder().sort(doc! { "internal_id": -1 }).build();
 
         Ok(self
             .get_job_collection()
             .find_one(filter, find_options)
             .await
-            .expect("Failed to fetch latest job by given job type"))
```

Comment:
expect would panic, miss from our side, let's remove all expects in the mongodb code